### PR TITLE
Move docs into subdirectory to make structor work

### PR DESCRIPTION
**Pull request by mrdrogdrog**
### Component/Part
Documentation

### Description
This PR moves the hole documentation files into a subdirectory that is in docs. Why?
Mkdocs can't build documentations for multiple versions. That's where structor comes in place. It checks out branches of a repository, builds the docs with mkdocs for this branch and patches the results together.
Structors looks for a requirements.txt file in every branch it processed. The requirements.txt includes every dependency that is needed by mkdocs (like themes, extension) to build the documentation for this branch.
And here comes the problem: The requirements.txt MUST be next to the mkdocs.yml. And structors searches for the mkdocs.yml in two places. In the root and in docs/.
So okay. Why don't we just let it like it was and move the requirements.txt and the mkdocs.yml into the docs folder? Easy answer: Because mkdocs wants the configuration (mkdocs.yml) next to the FOLDER that contains the documentation.

TL;DR. Convention over configuration sucks sometimes.

### Steps

- [x] Added / updated documentation
- [x] I read the [contribution documentation](https://github.com/hedgedoc/hedgedoc/blob/master/CONTRIBUTING.md) and signed-off my commits to accept the DCO.

### CODE REVIEW

**Review by DerMolly — CHANGES_REQUESTED** [review]
We should change the file names in dev/documentation.md

File: docs/content/dev/documentation.md [hunk 1]
```diff
@@ -15,7 +15,7 @@ To build the documentation locally you need to perform the following steps:
 
 1. make sure you have python3 installed
```
Comment (DerMolly) [inline docs/content/dev/documentation.md:16]: There should be a 
0. cd docs
?

**Review by DerMolly — CHANGES_REQUESTED** [review]

**Review by DerMolly — APPROVED** [review]

File: docs/content/dev/documentation.md [hunk 2]
```diff
@@ -4,18 +4,19 @@ Our documentation is build with [mkdocs](https://www.mkdocs.org).
...
 
 To build the documentation locally you need to perform the following steps:
 
-1. make sure you have python3 installed
+0. make sure you have python3 installed
+1. Go into the `docs` folder
 2. install all the dependencies possibly with a [venv](https://docs.python.org/3/library/venv.html):  
-`pip install -r mkdocs.requirements.txt`
+`pip install -r requirements.txt`
 3. start the mkdocs dev server `mkdocs serve` or let it build the documentation `mkdocs build`
```
Comment (ErikMichelson) [inline docs/content/dev/documentation.md:20]: As we have full sentences in this document and not only bullet-points, I'd suggest capitalizing the first letter of each point and end it with a dot.

File: docs/content/dev/documentation.md [hunk 3]
```diff
@@ -24,4 +25,4 @@ The documentation is deployed with [Messor Structor](https://github.com/traefik/
 
 The necessary Dockerfile and version menu template and also the github action to build the whole documentation can be found in the [docs.hedgedoc.org repo](https://github.com/hedgedoc/docs.hedgedoc.org). This repo is also used to deploy the actuall website to github.io.
 
-Messor Structor builds and deploys the documentation by finding all branches that follow the pattern `v*` and build each one seperatly via `mkdocs` by first installing `mkdocs.requirements.txt` and then building. Afterwards the menu go template is used to include a version switcher in the theme and make all the different versions available.
\ No newline at end of file
+Messor Structor builds and deploys the documentation by finding all branches that follow the pattern `v*` and build each one seperatly via `mkdocs` by first installing the dependencies from `requirements.txt` and then running mkdocs. Afterwards the menu go template is used to include a version switcher in the theme and make all the different versions available.
```
Comment (ErikMichelson) [inline docs/content/dev/documentation.md:28]:
```suggestion
Messor Structor builds and deploys the documentation by finding all branches that follow the pattern `v*`. For each branch the docs are generated seperatly by first installing the dependencies from `requirements.txt` and then running mkdocs. Afterwards the menu go template is used to include a version switcher in the theme and make all the different versions available.
```

The long "this is done by a for each done by b" sentence structure is quite hard to read. Maybe this sounds easier to understand?

**Review by ErikMichelson — CHANGES_REQUESTED** [review]

**Comment by mrdrogdrog** [comment]
btw: here is a demo: https://mrdrogdrog.github.io/hedgedoc-test/

**Review by ErikMichelson — APPROVED** [review]

**Review by davidmehren — APPROVED** [review]

File: README.md [hunk 4]
```diff
@@ -12,7 +12,7 @@ it by visiting our [HedgeDoc demo server][hedgedoc-demo].
 
 It is inspired by Hackpad, Etherpad and similar collaborative editors. This
 project originated with the team at [HackMD](https://hackmd.io) and now forked
-into its own organisation. [A longer writeup can be read in the history doc](docs/history.md).
+into its own organisation. [A longer writeup can be read in the history][history].
```
Comment (DerMolly) [inline README.md:15]: Well this links not really to a longer write-up, anymore. Maybe try to use the same text I used in `docs/content/index.md`?

**Review by DerMolly — CHANGES_REQUESTED** [review]

**Review by DerMolly — APPROVED** [review]
